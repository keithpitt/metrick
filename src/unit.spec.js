/* global describe, it, expect */
import Unit from './unit';
import * as index from './index';

const { Duration, duration: { hours, milliseconds } } = index;

describe('Unit', () => {
  it('uses a default multiplier of 1', () => {
    expect(new Unit().multiplier).toBe(1);
  });

  describe('in', () => {
    it('accepts a custom unit of the same type', () => {
      const myCustomDuration = new Duration(123);
      expect(() => 2::hours.in(myCustomDuration)).not.toThrow();
      expect(2::hours.in(myCustomDuration)).toMatchSnapshot();
    });

    it('accepts another unit of the same type', () => {
      expect(() => 2::hours.in(milliseconds)).not.toThrow();
      expect(2::hours.in(milliseconds)).toMatchSnapshot();
    });

    it('throws if used on a type other than a number', () => {
      expect(() => ({})::hours.in(milliseconds)).toThrowErrorMatchingSnapshot();
      expect(() => 'abc'::hours.in(milliseconds)).toThrowErrorMatchingSnapshot();
      expect(() => (() => 'hi')::hours.in(milliseconds)).toThrowErrorMatchingSnapshot();
    });

    it('throws if given a non-matching unit type', () => {
      const myWeirdUnit = new Unit(123);
      expect(() => 2::hours.in(myWeirdUnit)).toThrowErrorMatchingSnapshot();
    });
  });
});
