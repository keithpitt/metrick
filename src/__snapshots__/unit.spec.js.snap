exports[`Unit in accepts a custom unit of the same type 1`] = `58.53658536585366`;

exports[`Unit in accepts another unit of the same type 1`] = `7200000`;

exports[`Unit in throws if given a non-matching unit type 1`] = `"Duration.in: Passed unit must be the same type! Passed unit was a \`Unit\`."`;

exports[`Unit in throws if used on a type other than a number 1`] = `"Duration.in: Cannot be used on a non-number! We were given a \`object\`."`;

exports[`Unit in throws if used on a type other than a number 2`] = `"Duration.in: Cannot be used on a non-number! We were given a \`string\`."`;

exports[`Unit in throws if used on a type other than a number 3`] = `"Duration.in: Cannot be used on a non-number! We were given a \`function\`."`;
